File: BackEnd/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/Ordine.java
package it.unicam.cs.CasottoIdS.models;

import org.springframework.data.annotation.Id;

import java.util.List;

public class Ordine {
    @Id
    public String idOrdine;
    private String idUtente;
    private double costoTotale;
    private List<Prodotto> prodotti;
    private StatoOrdine stato;

    public Ordine(String idUtente, double costoTotale, List<Prodotto> prodotti) {
        this.idUtente = idUtente;
        this.costoTotale = costoTotale;
        this.prodotti = prodotti;
        this.stato = StatoOrdine.DA_GESTIRE;
    }

    public String getIdOrdine() {
        return idOrdine;
    }

    public String getIdUtente() {
        return idUtente;
    }

    public void setIdUtente(String idUtente) {
        this.idUtente = idUtente;
    }

    public double getCostoTotale() {
        return costoTotale;
    }

    public void setCostoTotale(double costoTotale) {
        this.costoTotale = costoTotale;
    }

    public List<Prodotto> getProdotti() {
        return prodotti;
    }

    public void setProdotti(List<Prodotto> prodotti) {
        this.prodotti = prodotti;
    }

    public StatoOrdine getStato() {
        return stato;
    }

    public void setStato(StatoOrdine stato) {
        this.stato = stato;
    }

    @Override
    public String toString() {
        return "Ordine{" +
                "idOrdine='" + idOrdine + '\'' +
                ", idUtente='" + idUtente + '\'' +
                ", costoTotale=" + costoTotale +
                ", prodotti=" + prodotti +
                ", stato=" + stato +
                '}';
    }
}

File: FrontEnd/Casotto/lib/models/Ordine.dart
import 'Prodotto.dart';
import 'StatoOrdine.dart';

class Ordine {
  late String idOrdine;
  late String idUtente;
  late String idOmbrellone;
  late double costoTotale;
  late List<Prodotto> prodotti;
  late StatoOrdine stato;

  Ordine(String idOrdine, String idUtente, String idOmbrellone,
      double costoTotale, List<Prodotto> prodotti, StatoOrdine stato) {
    this.idOrdine = idOrdine;
    this.idUtente = idUtente;
    this.idOmbrellone = idOmbrellone;
    this.costoTotale = costoTotale;
    this.prodotti = prodotti;
    this.stato = stato;
  }

  String getIdOrdine() {
    return idOrdine;
  }

  String getIdUtente() {
    return idUtente;
  }

  String getIdOmbrellone() {
    return idOmbrellone;
  }

  double getCostoTotale() {
    return costoTotale;
  }

  List<Prodotto> getListProdotti() {
    return prodotti;
  }

  String? getListaProdottiString(List<Prodotto> prodotti) {
    for (var prodottoObject in prodotti) {
      return prodottoObject.getNome();
    }
  }

  void setIdOrdine(String idOrdine) {
    this.idOrdine = idOrdine;
  }

  void setIdUtente(String idUtente) {
    this.idUtente = idUtente;
  }

  void setIdOmbrellone(String idOmbrellone) {
    this.idOmbrellone = idOmbrellone;
  }

  void setCostoTotale(double costoTotale) {
    this.costoTotale = costoTotale;
  }

  void setListProdotti(List<Prodotto> prodotti) {
    this.prodotti = prodotti;
  }

  StatoOrdine getStato() {
    return stato;
  }

  void setStato(StatoOrdine newStato) {
    this.stato = newStato;
  }

  String getStatoString() {
    switch (this.stato) {
      case StatoOrdine.DA_GESTIRE:
        return "DA GESTIRE";
      case StatoOrdine.IN_GESTIONE:
        return "IN GESTIONE";
    }
  }

  @override
  String toString() {
    return "Ordine{" +
        "idOrdine=" +
        idOrdine +
        ", idUtente=" +
        idUtente +
        ", prezzo=" +
        costoTotale.toString() +
        "${"Lista prodotti ordinati: $prodotti"}; " +
        "StatoOrdine:" +
        stato.toString();
  }
}
